elastic/eui
# feat(DragAndDrop): Switch from react-beautiful-dnd to its fork @hello-pangea/dnd

## Summary

`react-beautiful-dnd` is in maintenance mode and doesn't support React 18 or StrictMode. This PR fixes #6187 and switches to using a forked and maintained version of RBD - `@hello-pangea/dnd`. Both packages share the same API and are cross-compatible as long as EUI consumers aren't importing DND elements from both EUI and react-beautiful-dnd.

## QA

1. Pull the changes, run `yarn` and `yarn start`
2. Open http://localhost:8030/
3. Open drag and drop examples
4. Ensure they're working the same as on main

This code doesn't pass type checking because of differences in types between `@types/react@17` and `@types/react@18`. This will be fixed when #6827 is merged to `feature/react-18` and/or in following PRs.

### General checklist
